app/workers/domain_block_worker.rb

```diff
@@ -4,8 +4,11 @@ class DomainBlockWorker
   include Sidekiq::Worker
 
   def perform(domain_block_id, update = false)
-    BlockDomainService.new.call(DomainBlock.find(domain_block_id), update)
-  rescue ActiveRecord::RecordNotFound
-    true
+    domain_block = DomainBlock.find_by(id: domain_block_id)
+    if domain_block.nil?
```

Comment:
What's the benefit of rewriting it this way?

Reply:
The `ActiveRecord::RecordNotFound` error we want to catch is the one coming from the `DomainBlock` not existing, not one that could arise from a bug in `BlockDomainService`.

Reply:
In that case I would recommend using a guard clause, the return value `true` has no impact (in the rescue, it's just a way to avoid the don't supress exceptions code style check), to keep nesting level down.